test/cql-pytest/test_describe.py

```diff
@@ -1396,3 +1396,1170 @@ def try_create_role_with_salted_hash(role):
+                        LANGUAGE lua
+                        AS $$ return val1 + val2 $$""")
+        cql.execute(f"""CREATE AGGREGATE {ks}.my_aggregate(int)
+                        SFUNC my_udf
+                        STYPE int""")
+
+        [r1, r2, r3] = ["jack", "'b0b @nd d0b!'", "jane"]
+        cql.execute(f"CREATE ROLE {r1} WITH PASSWORD = 'pass1'")
+        cql.execute(f"CREATE ROLE {r2} WITH PASSWORD = 'pass2'")
+        cql.execute(f"CREATE ROLE {r3}")
+
+        cql.execute(f"GRANT {r1} TO {r3}")
+        cql.execute(f"GRANT {r3} TO {r2}")
+
+        cql.execute(f"GRANT ALL ON ALL KEYSPACES TO {r2}")
+        cql.execute(f"GRANT SELECT ON KEYSPACE {ks} TO {r3}")
+        cql.execute(f"GRANT MODIFY ON TABLE system.roles TO {r1}")
+        cql.execute(f"GRANT AUTHORIZE ON ALL ROLES TO {r1}")
+        cql.execute(f"GRANT DESCRIBE ON ALL ROLES TO {r1}")
+
+        [sl1, sl2] = ["my_service_level", "'s3rv!c3 l3v3l !!!'"]
+        cql.execute(f"CREATE SERVICE LEVEL {sl1} WITH TIMEOUT = 10ms AND WORKLOAD_TYPE = 'batch'")
+        cql.execute(f"CREATE SERVICE LEVEL {sl2} WITH TIMEOUT = 100s")
+
+        cql.execute(f"ATTACH SERVICE LEVEL {sl1} TO {r1}")
+        cql.execute(f"ATTACH SERVICE LEVEL {sl1} TO {r2}")
+        cql.execute(f"ATTACH SERVICE LEVEL {sl2} TO {r3}")
+
+        restore_stmts = list(cql.execute("DESC SCHEMA WITH INTERNALS AND PASSWORDS"))
+
+    def process_rows(rows: Iterable[DescRowType]) -> Iterable[DescRowType]:
+        def remove_empty_lines_from_rows(rows: Iterable[DescRowType]) -> Iterable[DescRowType]:
+            def aux(row: DescRowType) -> DescRowType:
+                result = row
+                result.create_statement = re.sub("\n+", "\n", result.create_statement)
+                return result
+
+            return map(aux, rows)
+
+        def remove_other_keyspaces(rows: Iterable[DescRowType]) -> Iterable[DescRowType]:
+            return filter(lambda row: row.type == ks or row.type == None, rows)
+
+        def remove_ids_from_mv_rows(rows: Iterable[DescRowType]) -> Iterable[DescRowType]:
+            def aux(row: DescRowType) -> DescRowType:
+                result = row
+                # ID is always the first option.
+                result.create_statement = re.sub("WITH ID = .+$^AND", "WITH", result.create_statement)
+                return result
+
+            return filter(lambda row: aux(row) if row.type == "view" else row, rows)
+
+        # The statements used for restoring the keyspaces are not sorted.
+        # Other keyspaces might've been created by other tests, so we exclude them here.
+        row_iter = remove_other_keyspaces(rows)
+        # See: scylladb/scylladb#20616.
+        row_iter = remove_ids_from_mv_rows(row_iter)
+        # Describing UDFs adds an additional newline character to the function's body.
+        # That inconsistency leads to failures when comparing the results of `DESCRIBE SCHEMA`.
+        row_iter = remove_empty_lines_from_rows(row_iter)
```

Comment:
Do we have an issue about this? This sounds a bit like a bug.

Reply:
It's not a bug. The newlines should not affect (as far as I can tell) the semantics of the UDF, so there should be no difference in the semantics. They're only very inconvenient for testing purposes. I've just created an issue for that: scylladb/scylladb#20711.

Reply:
Alternatively, maybe we could fix this within this PR

Reply:
It might be better indeed. I'll add a commit fixing it